processing/src/main/java/org/apache/carbondata/processing/loading/sort/impl/UnsafeParallelReadMergeSorterImpl.java

```diff
@@ -80,29 +80,25 @@ public void initialize(SortParameters sortParameters) {
   public Iterator<CarbonRowBatch>[] sort(Iterator<CarbonRowBatch>[] iterators)
       throws CarbonDataLoadingException {
     int inMemoryChunkSizeInMB = CarbonProperties.getInstance().getSortMemoryChunkSizeInMB();
-    UnsafeSortDataRows sortDataRow =
-        new UnsafeSortDataRows(sortParameters, unsafeIntermediateFileMerger, inMemoryChunkSizeInMB);
+    UnsafeSortDataRows[] sortDataRows = new UnsafeSortDataRows[iterators.length];
```

Comment:
Have you observed the memory foot print with new changes ?

Now as each thread uses it's own unsafeSortDataRows, each thread will allocate default 64 MB or configured value. So, local sort needs more memory after this changes !

Reply:
Isn't it worth it? 
If afraid of too many threads, we can limit the thread pool size and allocate the memory when it can run.

Reply:
i think, this number of threads depends upon iterators right, so based on iterators it will be done parallel right. or are you planning to define separate property for it?

Reply:
please check original code Line 91. 

As I said above, we can limit the pool size. We can use existing parameter from `sortParameters.getNumberOfCores()`